src/ensRegistry.ts

```diff
@@ -19,13 +19,34 @@ import {
 // Import entity types generated from the GraphQL schema
 import { Account, Domain, Resolver, NewOwner, Transfer, NewResolver, NewTTL } from './types/schema'
 
+function createDomain(node: string): Domain {
+  let domain = new Domain(node)
+  if(node == ROOT_NODE) {
+    domain = new Domain(node)
+    domain.owner = EMPTY_ADDRESS
+    domain.isMigrated = true
+  }
+  return domain
+}
+
+function getDomain(node: string): Domain|null {
+  let domain = Domain.load(node)
+  if(domain == null && node == ROOT_NODE) {
+    return createDomain(node)
+  }
+  return domain
+}
+
 // Handler for NewOwner events
-function _handleNewOwner(event: NewOwnerEvent, isMigrated:boolean): void {
+function _handleNewOwner(event: NewOwnerEvent, isMigrated: boolean): void {
   let account = new Account(event.params.owner.toHexString())
   account.save()
 
   let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
-  let domain = new Domain(subnode)
+  let domain = getDomain(subnode);
+  if(domain == null) {
```

Comment:
You have another `if(domain.name == null)` at line 51 so you can add it there rather than having own statement

Reply:
We're checking `if(domain == null)` here, not `domain.name`.

Reply:
ah sorry